GEOD-6 - Create RadioInterference Orika mapper

oehm-smith opened the pull request:
I did my own TimePeriodType and EffectiveDates converter though realised you may have already done one.  Let me know and I'll change this to use yours.

Similarly I did my own Date parsing (updated GMLDateUtils) and I think you were in the process of doing something different so I'll change to that also.

heyagosper commented on src/main/java/au/gov/ga/geodesy/support/mapper/orika/converters/ConverterTimePeriodTypeEffectiveDates.java:
```diff
@@ -0,0 +1,61 @@
+/**
+ * Created by brookes on 4/06/16.
+ */
+public class ConverterTimePeriodTypeEffectiveDates extends JAXBElementConverter<TimePeriodType, EffectiveDates> {
```
naming convention?

heyagosper commented on src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java:
```diff
@@ -0,0 +1,74 @@
+import java.util.TimeZone;
+
+/**
+ * Tests the mapping of a GeodesyML humiditySensor element
```
copy-paste doco fail ;-)

oehm-smith commented:
Feedback commit.  Date handling improved.

heyagosper commented on src/test/java/au/gov/ga/geodesy/support/TestResources.java:
```diff
@@ -38,6 +38,20 @@ private static String geodesyMLSiteLogsDirectory() {
+    /**
+     * GeodesyML site logs Test Data directory (files with modifications) relative to classpath root.
+     */
+    private static String geodesyMLTestDataSiteLogsDirectory() {
```
the good thing about having the testData (or "custom" or whatever we call it) inside the geodesyml is it does not need a unique name, and does not need a new method. that is, you can access a customised sopac file using 
TestResources.sopacSiteLogReader("testData/MOBS")
and a custom geodesyml file using 
TestResources.geodesyMLSiteLogReader("testData/MOBS")

oehm-smith replied:
Yeah.  Lets tackle this as separate work later.

oehm-smith commented:
Hold off on merging this.  I should be able to fix up the Converters.

oehm-smith commented:
I've updated it to use the same code Heya used in one of her mappers.  Simpler.  Tests are still quite extensive but pass of courses.  I changed to use Hamcrest in tests.

heyagosper commented on src/test/java/au/gov/ga/geodesy/support/mapper/dozer/TranslateTest.java:
```diff
@@ -700,13 +700,13 @@ public void testZIMJ_RadioInterference_WithEffectiveDates_NoToForms(GeodesyMLTyp
-        Assert.assertEquals(0, end.getValue().size());
+        Assert.assertNull(end);
+
+        Assert.assertEquals(
```
you may as well change this one to ham and cress while you're there. no biggie though

heyagosper commented on src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java:
```diff
@@ -50,16 +53,16 @@ public void testMappingWithoutToDate() throws Exception {
         // IRKJ_RadioInterference_NoToDate has no end/to date
-        Assert.assertEquals(((TimePeriodType) radioInterferencesDTO.getValidTime().getAbstractTimePrimitive().getValue()).getEndPosition().getValue().size(), 0);
-        Assert.assertNull(radioInterferenceEntity.getEffectiveDates().getTo());
+        MatcherAssert.assertThat(((TimePeriodType) radioInterferencesDTO.getValidTime().getAbstractTimePrimitive().getValue()).getEndPosition().getValue().size(), Is.is(0));
```
my reading was that "Is.is" is identical to "IsEqual.equalTo" is that right? your way saves a few chars of code, but I like the expressiveness of the other way (implies that is using the equals method and not comparing references for example).
Not saying this should be changed just discussing

heyagosper commented on src/test/java/au/gov/ga/geodesy/support/mapper/dozer/TranslateTest.java:
```diff
@@ -700,13 +700,13 @@ public void testZIMJ_RadioInterference_WithEffectiveDates_NoToForms(GeodesyMLTyp
+                GMLDateUtils.stringToDateToStringMultiParsers(begin.getValue().get(0)),
+                "2012-01-15T00:00:00.000Z");
+//        Assert.assertNotNull(end);
+//        Assert.assertEquals(0, end.getValue().size());
```
remove instead of commenting out

heyagosper commented on src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java:
```diff
@@ -97,29 +100,29 @@ public void testMappingWithToDate() throws Exception {
             String actualFromDateString = ((TimePeriodType) radioInterferencesDTO.getValidTime().getAbstractTimePrimitive().getValue()).getBeginPosition().getValue().get(0);
             String actualFromDateFormatted = GMLDateUtils.stringToDateToString(actualFromDateString, format);
-            Assert.assertEquals(expectedFromDateFormatted, actualFromDateFormatted);
+            MatcherAssert.assertThat(expectedFromDateFormatted, Is.is(actualFromDateFormatted));
```
yeah I agree about the annoyance of using the static method imports

oehm-smith commented:
Ok good to go now.  It now includes some Sonar requested changes.

lbodor commented on src/test/java/au/gov/ga/geodesy/support/utils/ToFromDate.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Represents the direction of mapping, for example XML types to JPA entities
+ * or vice versa.
+ */
```
Incorrect comment. I don't think that this enum type adds any value. It is only used to abstract over the choice of from date or to date of an effective dates pair. More straightforward code can be written without this type. I suggest we remove it.

oehm-smith commented:
NOPE, forgot to incoropate into the larger mapper!

oehm-smith commented:
Closing and doing a new PR
